src/pages/index.js

```diff
@@ -74,54 +103,83 @@ const cardTitleInput = addCardFormElement.querySelector(
 );
 const cardURLInput = addCardFormElement.querySelector(".modal__input_type_url");
 
-function closeWithEsc(event) {
-  if (event.key === "Escape") {
-    const modal = document.querySelector(".modal_opened");
-    closeModal(modal);
-  }
-}
-
-function closeModalOnRemoteClick(event) {
-  if (event.target === event.currentTarget) {
-    closeModal(event.currentTarget);
-  }
-}
-
-function openModal(modal) {
-  // add class to modal
-  document.addEventListener("keydown", closeWithEsc);
-  modal.addEventListener("mousedown", closeModalOnRemoteClick);
-  modal.classList.add("modal_opened");
-}
-
-function closeModal(modal) {
-  //remove class from modal
-  document.removeEventListener("keydown", closeWithEsc);
-  modal.removeEventListener("mousedown", closeModalOnRemoteClick);
-  modal.classList.remove("modal_opened");
-}
+const pictureURLInput = pictureFormElement.querySelector(
+  ".modal__input_type_url"
+);
 
 function renderCard(cardData) {
   const card = createCard(cardData);
   section.addItem(card);
-  //wrapper.prepend();
-  //Section.renderItems(item);
 }
 
 function handleImagePreview(cardData) {
-  popupImage.open(cardData); 
+  popupImage.open(cardData);
+}
+
+function handleSubmit(request, popupInstance, form, loadingText = "Saving...") {
+  popupInstance.renderLoading(true, loadingText);
+  // setTimeout(() => {
+  request()
+    .then(() => {
+      popupInstance.close();
+      if (form) {
+        form.reset();
+      }
+    })
+    .catch(console.error)
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+  // }, 1000);
 }
 
 function handleProfileFormSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues);
-  editProfilePopup.close();
-  editProfileForm.reset();
+  function makeRequest() {
+    return api.setUserInfo(inputValues).then((userData) => {
+      userInfo.setUserInfo(userData);
+    });
+  }
+  handleSubmit(makeRequest, editProfilePopup, editFormValidator);
 }
 
 function handleAddCardFormSubmit({ name, link }) {
-  renderCard({ name, link }, cardsWrap);
-  addProfilePopup.close();
-  addCardForm.reset();
+  function makeRequest() {
+    return api
+      .addCard({ name, link })
+      .then((res) => {
+        renderCard(res, cardsWrap);
+      })
+      .catch((error) => {
+        console.error("Error adding card:", error);
+      });
+  }
+
+  handleSubmit(makeRequest, addCardPopup, addCardValidator);
+}
+
+function handleDeleteCardFormSubmit(card) {
+  api
+    .removeCard(card._id)
+    .then((res) => {
+      card._handleDeleteButton();
```

Comment:
`_handleDeleteButton`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.